ui/src/app/components/inputs.tsx

```diff
@@ -127,6 +128,52 @@ export const CheckBox = ({onChange, ...props}) => {
   />;
 };
 
+interface LabeledCheckboxProps {
+  initialValue: boolean;
+  disabled?: boolean;
+  onChange?: Function;
+  style?: object;
+  checkboxStyle?: object;
+  labelStyle?: object;
+  label: string;
+}
+
+interface LabeledCheckboxState {
+  value: boolean;
+}
+
+export class LabeledCheckbox extends React.Component<LabeledCheckboxProps, LabeledCheckboxState> {
+  constructor(props: any) {
+    super(props);
+    this.state = {
+      value: props.initialValue
+    };
+  }
+
+  toggleValue() {
+    if (!this.props.disabled) {
+      this.setState(previousState => ({value: !previousState.value}));
+    }
+  }
+
+  render() {
+    return <div style={this.props.style}>
+      <CheckBox
+          style={{...this.props.checkboxStyle, verticalAlign: 'middle'}}
+          checked={this.state.value}
+          disabled={this.props.disabled}
+          onChange={e => this.props.onChange(e)}
```

Comment:
onChange is nullable in the props typedef, so I think this needs to check first whether this.props.onChange exists.

Reply:
Good catch, thank you.